### ISSUE TITLE
micro_continual: MicroStreamConfig accepts n_steps that overflows signed int32, silently corrupting the regime schedule

### ISSUE THREAD

**Issue by rama0x1** [SlopDotCash/asi#792]
## Summary

`MicroStreamConfig.__post_init__` in `alberta_framework/benchmarks/micro_continual.py`
validates that `n_regimes`, `regime_length`, `dim`, `n_classes`, and
`n_components` are each a positive Python `int`, but never bounds the
*derived* `n_steps = n_regimes * regime_length` against the signed int32
domain. `micro_continual` is the CLI-exposed "prototype on the micro suite"
lane (`repository-contract.md`'s open-lane doc, and directly reachable via
`--n-regimes`/`--regime-length` on `python -m
alberta_framework.benchmarks.micro_continual run|ladder`), so this is
user/automation-reachable, not just a theoretical input.

`stream()` (the function every arm and the transfer-validation ladder
consume) then does:

```python
regime_index = (
    jnp.arange(n_steps, dtype=jnp.int32) // config.regime_length
).astype(jnp.int32)
```

For `n_steps > 2**31 - 1` this stores per-step index values in a `jnp.int32`
buffer that cannot represent them. Casting an out-of-range Python/NumPy int
to `int32` truncates with two's-complement wraparound rather than raising —
confirmed directly:

```pycon
>>> import numpy as np
>>> np.int32(2**31)      # one past INT32_MAX
-2147483648
```

So a config such as `n_regimes=60_000, regime_length=60_000` (`n_steps =
3_600_000_000`, i.e. `~1.68x` over `INT32_MAX = 2_147_483_647`) constructs
without error today, and every regime-index value computed for steps past
the wraparound boundary would be silently wrong (negative), corrupting the
regime schedule — and therefore the measured stream — instead of failing
closed.

This is exactly the resource/scalar-contract gap this repository has been
closing across other benchmark/evaluation configs this cycle (e.g.
`alberta_framework/benchmarks/upgd_ipmnist.py`'s `IPMNISTConfig` already
guards the analogous `n_tasks * task_length > _INT32_MAX` product with an
explicit `ValueError`; `micro_continual.py`'s sibling config never got the
same guard).

## Reproduction

```bash
.venv/bin/python -c "
from alberta_framework.benchmarks.micro_continual import MicroStreamConfig
cfg = MicroStreamConfig(family='input_permutation', n_regimes=60_000, regime_length=60_000)
print(cfg.n_steps, cfg.n_steps > 2**31 - 1)  # 3600000000 True -- constructs today, should raise
"
```

## Fix

Add the same `n_regimes * regime_length > _INT32_MAX` bound check
`upgd_ipmnist.IPMNISTConfig` already applies, raising `ValueError` at
construction (before any array is built) instead of allowing the silent
int32 wraparound downstream.

## Scope

`alberta_framework/benchmarks/micro_continual.py` only (my assigned
territory: `alberta_framework/{benchmarks,evaluation,utils}/`). One
variable: bound the one derived product that overflows int32 today.
